rs/nns/integration_tests/src/cycles_minting_canister.rs

```diff
@@ -514,8 +542,8 @@ fn test_cmc_cycles_create_with_settings() {
         10_000_000_000_000,
     )
     .unwrap();
-    let status = canister_status(&state_machine, *TEST_USER1_PRINCIPAL, canister);
```

Comment:
Why is this change necessary?

Reply:
To add context, I guess I'm asking - what's changing about the state machine API?

Reply:
I've updated the PR description to provide more background on why the changes of the tests are necessary. In this particular case, we have a canister with no controllers and thus the `canister_status` call fails. Before this PR, the error message would come from the execution of the message and it would contain an explicit list of canister controllers whose content was asserted in the test. With the change from this PR, the error message comes from the ingress filter (as it'd do in production) and thus it is less verbose and the set of controllers cannot be derived from the new error message => we use the `canister_info` endpoint that can be called by an arbitrary (proxy) canister.